cmd/cluster-agent/subcommands/compliance/command.go

```diff
@@ -32,7 +32,8 @@ func Commands(globalParams *command.GlobalParams) []*cobra.Command {
 	}
 
 	// TODO: The SecAgent Check package should be a component
-	complianceCmd.AddCommand(check.CommandsWrapped(bundleParams)...)
+	// The globalParams input should always be nil here. This is used to differentiate between entering from Cluster-Agent or the Security-Agent
```

Comment:
Passing `nil` to tell if the function is called from the Cluster Agent is not explicit from the point of view of the implementation of `CommandsWrapped`.
What is the expected differences of behaviour for  `CommandsWrapped` when calling from the Cluster or the from Security Agent?
My understanding is that `CommandsWrapped` should override the logger and the configuration for the cluster Agent.

Reply:
Yea, even if this is pending refactoring, I should make it more explicit. I added a flag to determine if it's coming from cluster-agent or security-agent. CommandsWrapped uses either the params from the security-agent to instantiate the logger and config, or the whole bundle param from cluster-agent.

Reply:
You can use a lambda function to configure `BundleParams`.
Note:  The values inside `globalParams` are not yet set when calling `SecurityAgentCommands` but can safely used inside `RunE`. 
```go
func SecurityAgentCommands(globalParams *command.GlobalParams) []*cobra.Command {
	return commandsWrapped(func() core.BundleParams {
		return core.BundleParams{
			ConfigParams: config.NewSecurityAgentParams(globalParams.ConfigFilePaths),
			LogParams:    log.LogForOneShot(command.LoggerName, "info", true),
		}
	})
}

func ClusterAgentCommands(bundleParams core.BundleParams) []*cobra.Command {
	return commandsWrapped(func() core.BundleParams {
		return bundleParams
	})
}

func commandsWrapped(bundleParamsFactory func() core.BundleParams) []*cobra.Command {
	checkArgs := &CliParams{}

	cmd := &cobra.Command{
		Use:   "check",
		Short: "Run compliance check(s)",
		Long:  ``,
		RunE: func(cmd *cobra.Command, args []string) error {
			checkArgs.args = args
			bundleParams := bundleParamsFactory()

			if checkArgs.verbose {
				bundleParams.LogParams = log.LogForOneShot(bundleParams.LogParams.LoggerName(), "trace", true)
			}
  / ...
}
```